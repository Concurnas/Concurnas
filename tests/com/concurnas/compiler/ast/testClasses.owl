//define classes, interfaces et al
d=s;
class MyClass<X>(-n String, -m String)
{
	a =6;
	def add(toAdd X) void	{	}
	
	def do(n int) void
	{
		this.a= n**5;
	}
	
	class Inner
	{
	
	}
}

class Singleton
{
}

zzz = new A.C(2,3);

bobby List<Integer> = new List<Integer>(1);

a = actor MyClass<String>(3);

stuff = zzz.a
stuff = zzz\.a
