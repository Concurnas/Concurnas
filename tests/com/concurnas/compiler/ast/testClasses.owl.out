{
d = s ;
class MyClass < X > ( protected -n String , protected -m String ) {
a = 6 ;
def add ( var toAdd X ) void {
}
def do ( var n int ) void {
this . a = n ** 5 ;
}
class Inner {
}
}
class Singleton {
}
zzz = new A.C ( 2 , 3 ) ;
bobby List < Integer > = new List < Integer > ( 1 ) ;
a = new com.concurnas.lang.TypedActor < MyClass < String > > ( 3 ) ;
stuff = zzz . a ;
stuff = zzz \. a ;
}